### ISSUE TITLE
Passing Parameter value to the report before being rendered

### ISSUE THREAD

**Issue by robertldecastro** [DevExpress-Examples/reporting-web-document-viewer-in-javascript-with-vue#24]
Has anybody tried setting value to an invisible parameter to the report before being rendered?

Tried the example from

"https://docs.devexpress.com/XtraReports/js-DevExpress.Reporting.Viewer.JSReportViewer#js_devexpress_reporting_viewer_jsreportviewer_getparametersmodel"

but it doesn't work

this.previewModel.GetParametersModel()["ProductId"](1)
            this.previewModel().StartBuild()

Console Message:

vue.runtime.esm.js?2b0e:619 [Vue warn]: Error in mounted hook: "TypeError: this.previewModel.GetParametersModel is not a function"

found in

---> <WebDocumentViewer> at src/views/ReportViewer.vue
       <VMain>
         <VApp>
           <WebDocumentViewer> at src/views/Index.vue
             <App> at src/App.vue
               <Root>

**Comment by ZhannaGaidukova** [comment]
Hi @robertldecastro,
From what I gather, you tried to access the client-side parameters model in the client-side `BeforeRender` event. 

The point is that the viewer doesn't yet initialize the parameters model at this time. Therefore, accessing a parameter model before the viewer being rendered won't work.

If you wish to preset report parameter values before the preview appears, try the [ParametersInitialized](https://docs.devexpress.com/XtraReports/DevExpress.AspNetCore.Reporting.WebDocumentViewer.WebDocumentViewerClientSideEventsBuilder.ParametersInitialized(System.String)?p=netframework) event, which is available with **v21.1**. 
In this event handler, you can access a parameter and specify its `value`. Don't forget to call the `Submit` function and initiate document building. 

For example:
```
function onParameterssInitialized(s, e) {
        var p1 = e.ActualParametersInfo.filter(x => x.parameterDescriptor.name == "parameter1")[0];
        p1.value(100);
        p1 && e.Submit();
}
```
More information is available in the [Remarks](https://docs.devexpress.com/XtraReports/DevExpress.AspNetCore.Reporting.WebDocumentViewer.WebDocumentViewerClientSideEventsBuilder.ParametersInitialized(System.String)?p=netframework#remarks) section.

Should you have additional questions or require further assistance, please feel free to contact the DevExpress Support Team at https://www.devexpress.com/ask.

**Comment by robertldecastro** [comment]
Thank you Zhanna.

attached it to the "callbacks" parameter of the viewerOption and it worked.

var beforeRender = {
               ParametersInitialized(s, e) {
               //s.GetPreviewModel().zoom(1)
               var p1 = e.ActualParametersInfo.filter(x => x.parameterDescriptor.name == "parameter1")[0];
                p1.value("testvalue");
                p1 && e.Submit();
            }
          }

 var viewerOptions = {
                reportUrl: "TestReport", // The URL of a report.
                callbacks: beforeRender,
                requestOptions: {
                host: "https://localhost:8080/",
                invokeAction: "DXXRDV"
                }
            }
